lib/cli.rb

```diff
@@ -2,240 +2,275 @@
+        parser.separator ''
+        define_options(parser)
+        parser.on_tail('-h', '--help', 'Show this message') do
+          puts parser
+          exit
         end
       end
+    end
 
-      # rubocop:disable Metrics/MethodLength
-      def define_options(parser)
-        @force = false
-        @skip_client = false
-        @drivers = []
-        @debug = false
+    def define_options(parser)
+      parser.on('--debug', TrueClass, 'Enable debug mode',
+                &method(:debug=))
 
-        parser.on('--debug', TrueClass, 'Enable debug mode',
-                  &method(:debug=))
+      parser.on('-p', '--platform <platform_name>', String,
+                'Install VM for specified platform',
+                &method(:platform=))
 
-        parser.on('-p', '--platform <platform_name>', String,
-                  'Install VM for specified platform',
-                  &method(:platform=))
+      parser.on('-f', '--force', TrueClass,
+                'Install all VM, replace studio if exist',
+                &method(:force=))
 
-        parser.on('-f', '--force', TrueClass,
-                  'Install all VM, replace studio if exist',
-                  &method(:force=))
+      parser.on('--skip_client', TrueClass,
+                'Skip client images installation',
+                &method(:skip_client=))
 
-        parser.on('--skip_client', TrueClass,
-                  'Skip client images installation',
-                  &method(:skip_client=))
+      parser.on('-d', '--drivers <drivers_list>', Array,
+                'List of driver attach in install',
+                &method(:drivers=))
 
-        parser.on('-d', '--drivers <drivers_list>', Array,
-                  'List of driver attach in install',
-                  &method(:drivers=))
+      parser.on('--driver-path <driver_path>', String,
+                'Path to the location of the driver wanted to be installed',
+                &method(:driver_path=))
+    end
+  end
 
-        parser.on('--driver-path <driver_path>', String,
-                  'Path to the location of the driver wanted to be installed',
-                  &method(:driver_path=))
-      end
-      # rubocop:enable Metrics/MethodLength
+  # class CLI
+  class CLI < T::Struct
+    extend AutoHCK::Models::JsonHelper
+    extend T::Sig
+
+    attr_reader :install, :mode
+    attr_accessor :common, :test
+
+    def initialize
+      super(
+        test: TestOptions.new,
+        common: CommonOptions.new,
+        install: InstallOptions.new
+      )
+
+      @sub_parser = {
+        'test' => @test.create_parser,
+        'install' => @install.create_parser
+      }
+
+      @parser = @common.create_parser(@sub_parser)
     end
 
     def parse(args)
       left = @parser.order(args)
       @mode = left.shift
       @sub_parser[@mode]&.order!(left) unless @mode.nil?
+
+      restore_session if @test.latest_session || @test.session
```

Comment:
Instead of passing self to `AutoHCK::Session.load` and letting it modify, let's pass the path to the session.
Such a method may be called from `bin/auto_hck` and return an instance of loaded `CLI`, which will replace the whole existing `CLI` instance.

So instead of having these lines in `session.rb`:
```rb
      cli.test = session.test
      cli.common = session.common
```

...add a line like the following in `auto_hck`:
```rb
cli = Session.load(session_path)
```

You don't have to list each attribute of `CLI` this way.